steps/25_pretrain/step.md

```diff
@@ -0,0 +1,244 @@
+# API for pre-training and fine-tuning
+
+Contributors: @fkiraly
+
+## High-level summary 
+
+### The Aim
+
+`sktime` now has a number of estimators that can do pre-training, fine-tuning, global learning, and cross-learning.
+
+However, the current API design for these use cases has a few problems, and repeatedly issues have been opened for a rework.
+
+This STEP is about finalizing a good interface for:
+
+* global forecasting
+* pre-training
+* fine-tuning of foundation models
+* zero-shot use of foundation models
+
+References:
+
+* conceptual design issue: https://github.com/sktime/sktime/issues/6580
+* umbrella issue foundation models: https://github.com/sktime/sktime/issues/6177
+* newer issue: https://github.com/sktime/sktime/issues/7838
+
+### requirements
+
+* design covers the above use cases with a simple interface
+* composability - use of sensible pipelines, tuning, etc should be simple and not require major surgery in current compositors
+* downwards and upwards compatibility - design should not impact current extension contracts
+* maintainability: maintaining the framework and estimators with the above capabilities should be simple
+
+### The proposed solution
+
+Our proposed solution adds a new state, and a simple switch.
+
+No new public methods are added beyond this, and signatures of methods are not modified.
+
+Estimators get a third state, "pretraining phase", besides unfitted and fitted.
+
+The solution is best illustrated in the basic vignette below.
+
+### Discussion of current solutions
+
+There are multiple current solutions, all have problems:
+
+#### Global forecasting
+
+Forecasters inheriting from `_BaseGlobalForecaster`.
+
+A `y` is added in the `predict` methods. If this is passed, the `fit` is interpreted
```

Comment:
Can you explain why this is interpreted as pertaining pass? And do you want to distinguish between pretraining and fine-tuning on a technical level?

Reply:
I do not think there is - or that there should be - a significant distinction between pretraining and fine-tuning on the API level.

I am aware that fine-tuning is not a training epoch on the entire network, usually, but it can be separated in the API as "pre-`__init__`" and "pre-`fit`".

Reply:
Perhaps, this need then some clarification what is meant here. In the current design passing `y` in `predict` only means that this is the context now and no context from the `fit` call is used. 

So I think technically setting `y` in predict does not change anything how `fit` should be interpreted.

Maybe, I do not fully understand this part of the STEP and this leads to confusions

Reply:
> Perhaps, this need then some clarification what is meant here. In the current design passing `y` in `predict` only means that this is the context now and no context from the `fit` call is used. 

Imo this is exactly the problem! In the "vanilla" `fit`, the input is always the "context", in transformers speak. Changing the interface that the context is in `predict`, and only there, would break interface uniformity.